src/components/common/ColonyActions/ActionsListItem/config/helpers.tsx

```diff
@@ -0,0 +1,146 @@
+import React from 'react';
+import Decimal from 'decimal.js';
+
+import Numeral from '~shared/Numeral';
+import { Colony, FormattedAction } from '~types';
+import { formatRolesTitle } from '~utils/colonyActions';
+import { findDomain } from '~utils/domains';
+import {
+  getFormattedTokenValue,
+  getTokenDecimalsWithFallback,
+} from '~utils/tokens';
+import FriendlyName from '~shared/FriendlyName';
+
+import styles from '~shared/ListItem/ListItem.css';
+
+const formatReputationChange = (decimals: string, reputationChange?: string) =>
+  getFormattedTokenValue(
+    new Decimal(reputationChange || '0').abs().toString(),
+    decimals,
+  );
+
+export const mapItemToExpectedFormat = (
+  item: FormattedAction,
+  colony?: Colony,
+) => {
+  const formattedRolesTitle = formatRolesTitle(item.roles); // @TODO: item.actionType === ColonyMotions.SetUserRolesMotion ? updatedRoles : roles,
+  const reputationChange = formatReputationChange(
+    item.decimals,
+    item.reputationChange,
+  );
+  return {
+    ...item,
+    amount: (
+      <Numeral
+        value={item.amount} // @TODO: getAmount(item.actionType, item.amount)
+        decimals={getTokenDecimalsWithFallback(item.decimals)}
+      />
+    ),
+    direction: formattedRolesTitle.direction,
+    fromDomain: findDomain(item.fromDomain, colony)?.name,
+    initiator: (
+      <span className={styles.titleDecoration}>
+        <FriendlyName user={item.initiator} autoShrinkAddress />
+      </span>
+    ),
+    recipient: (
+      <span className={styles.titleDecoration}>
+        <FriendlyName user={item.recipient} autoShrinkAddress />
+      </span>
+    ),
+    reputationChangeNumeral: <Numeral value={reputationChange} />,
+    reputationChange,
+    rolesChanged: formattedRolesTitle.roleTitle,
+    toDomain: findDomain(item.toDomain, colony)?.name,
+  };
+
+  /*
+
+Note that the following transformations also exist in the Dapp. Because they are async,
```

Comment:
I've left this comment here as a reminder of outstanding transformations that hopefully will be handled by the cdapp equivalent of the actions resolver.

Can't do anything in this pr since we're using mock data.

Reply:
Local resolvers are [a deprecated feature](https://www.apollographql.com/docs/react/local-state/local-resolvers/), so custom hooks sound like the best next option

Reply:
@jakubcolony Or have all/most of the transformation handled by the block ingestor? I.e. it picks up the on-chain event and adds to the db an actions item that's as close as possible (minus the jsx) to the format expected by the actions list item. (E.g. providing a User object instead of an address string). Then we query the db for that action to display in the list. 

Isn't that more or less the flow we're expecting to implement anyway? 
cdapp => on-chain event => block ingestor => db => cdapp

Reply:
That's kinda the general idea of the block investor anyway, to handle as much data processing as possible, so we can offload it from the client.

Granted, this is not possible in all cases, but where that's the case, we should strive to achieve that.